core-data/reducer.js

```diff
@@ -42,6 +42,26 @@ export function terms( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing authors state. Keyed by id.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function authors( state = [], action ) {
+	switch ( action.type ) {
+		case 'RECEIVE_AUTHORS':
+			return [
+				...state,
+				...action.authors,
+			];
```

Comment:
I believe this would create duplicate authors if we trigger the request twice (it's not happening at the moment though :)). Maybe we can store them by Id :).

Also, why do we have an `authors` reducer instead of a `users` reducer. I believe we should have a `users` reducer and use the `getAuthors` selector to filter the data according to the user's object. This way we won't be duplicating data when we introduce the `users` fetching.

This looks more complex at first sight, but it's the way to go. Just imagine how you'd update the authors list if you create a new author from the JS side for instance? If it's all handled in a generic way in a `users` reducer, the selector will update its changes without any specific behavior to trigger the request again.

Reply:
> I believe this would create duplicate authors if we trigger the request twice (it's not happening at the moment though :)). Maybe we can store them by Id :).

Fixed in 97a7a351

> Also, why do we have an `authors` reducer instead of a `users` reducer. I believe we should have a `users` reducer and use the `getAuthors` selector to filter the data according to the user's object.

Because of #6361. With this approach, we keep the definition of "which user is an author" server-side.

Reply:
> While this may be ok shortterm, I don't think it's good because we have to think of the `core-data` module as a generic module to handle client data in any WP Admin context.

Would you like me to move it to editor data then?

> How can we move this knowledge client-side? what property do we need in the `user`'s object to say that it's an author or not?

We'd need to run a second query client-side for every user collection request. First, a `WP_User_Query( [ 'who' => 'authors' ] )` to get all authors, and the second `WP_User_Query` for the full query. When preparing the results for return, we'd inspect whether the user is a part of the first query and include an author attribute correspondingly.

My preference is to keep this current implementation.

Reply:
In fact, this current implementation can be progressively enhanced to a better implementation in the future, should one arise.

Reply:
So we discussed this with @aduth and I believe there's a compromise to find here if we can't move the "author" knowledge to the client.

we can track the `?who=authors` response as a set of IDs pointing to the canonical users state.

Also this relates to #6395 helper since `?who=authors` can be considered just one `query` on the `user` model.

I'm fine using a compromise for now and generalizing this as a query in #6395 or similar.

Reply:
Can you clarify what changes you'd like me to make to this PR? Or, open a PR against this PR with your suggested changes?

Reply:
Sure, give me some minutes

Reply:
Here https://github.com/WordPress/gutenberg/pull/6522